### PR TITLE
Add validation for provider config

### LINKED ISSUE #22: Validate config
All config should go through a validation stage and respond with good error messages. Especially important is the provider contains throws unreadable errors if there are connection issues.

### PR DESCRIPTION
**Pull request by ali2251**
Fixes #22

### CODE REVIEW

File: src/ts/provider.ts [hunk 1]
```diff
@@ -12,12 +16,32 @@ const getGanacheProvider = () => {
...
 
     return infuraProvider
 }
+
+
+export async function validateProviders(local : boolean = false) {
+    let provider = 0
+    try {
+        if (local)  {
+            await getInfuraProvider().getTransactionCount("0xD115BFFAbbdd893A6f7ceA402e7338643Ced44a6")
```
[the remark this answers is not in the document]
Comment (ali2251) [reply src/ts/provider.ts]: added

File: src/ts/provider.ts [hunk 2]
```diff
@@ -12,12 +16,32 @@ const getGanacheProvider = () => {
     return ganache;
 };
 
+export const getJsonRPCProvider = () => {
+    return new ethers.providers.JsonRpcProvider(config.jsonRpcUrl)
```
[the remark this answers is not in the document]
Comment (ali2251) [reply src/ts/provider.ts]: agreed

File: src/ts/startUp.ts [hunk 3]
```diff
@@ -3,22 +3,35 @@ import { ethers } from "ethers";
...
+import { getJsonRPCProvider, validateProviders } from "./provider";
 const config = require("./config.json") as IConfig;
-const provider = new ethers.providers.JsonRpcProvider(config.jsonRpcUrl);
-provider.pollingInterval = 100;
 
 
-const watcherWallet = new ethers.Wallet(config.watcherKey, provider);
-const watcher = new KitsuneWatcher(provider, watcherWallet);
-const inspector = new KitsuneInspector(10, provider);
+validateProviders(true).then((res) => {
```
[the remark this answers is not in the document]
Comment (ali2251) [reply src/ts/startUp.ts]: agreed

File: src/ts/provider.ts [hunk 4]
```diff
@@ -12,12 +16,32 @@ const getGanacheProvider = () => {
...
+    try {
+        if (local)  {
+            await getInfuraProvider().getTransactionCount("0xD115BFFAbbdd893A6f7ceA402e7338643Ced44a6")
+            ++provider;
+            await getJsonRPCProvider().getTransactionCount("0xD115BFFAbbdd893A6f7ceA402e7338643Ced44a6")
+        } else {
+            await getInfuraProvider().getTransactionCount("0xD115BFFAbbdd893A6f7ceA402e7338643Ced44a6")
+        }
+        
+    } catch(err) {
```
[the remark this answers is not in the document]
Comment (ali2251) [reply src/ts/provider.ts]: updated it now such that numbers or names are not required

**Comment by yahgwai** [comment]
Hey Ali, are you waiting for anything from me on this?

**Pull request merged**